### PR TITLE
[docs] Why Tune?

### PR DESCRIPTION
**Pull request by richardliaw**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: doc/source/tune/why_tune.rst [hunk 1]
```diff
@@ -0,0 +1,52 @@
+Why choose Tune?
```
Comment (anabranch) [inline doc/source/tune/why_tune.rst:1]: Can we just add this as a section in the overview instead of a new file?

**Review by bllchmbrs — DISMISSED** [review]

**Comment by richardliaw** [comment]
cc @krfricke

File: doc/source/tune/why_tune.rst [hunk 2]
```diff
@@ -0,0 +1,30 @@
+Tune offers cutting-edge optimization algorithms.
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+As a user, you're probably looking into hyperparameter optimization because you want to quickly increase your model performance.
+
+Tune enables you to leverage a variety of these cutting edge optimization algorithms, reducing the cost of tuning by `aggressively terminating bad hyperparameter evaluations <tune-scheduler-hyperband>`_, intelligently :ref:`choosing better parameters to evaluate <tune-search-alg>`, or even :ref:`changing the hyperparameters during training <tune-scheduler-pbt>` to optimize hyperparameter *schedules*.
```
Comment (sumanthratna) [inline doc/source/tune/why_tune.rst:6]: nit: maybe make this a bullet list? this isn't a big deal, though

File: doc/source/tune/why_tune.rst [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+Tune simplifies your workflow.
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A key problem with machine learning frameworks is the need to restructure all of your code to fit the framework.
+
+With Tune, you can optimize your model just by :ref:`adding a few code snippets <tune-tutorial>`.
+
+Further, Tune actually removes boilerplate from your code training workflow, automatically :ref:`managing checkpoints <tune-checkpoint>` and :ref:`logging of results to tools <tune-logging>` such as MLFlow and TensorBoard.
```
Comment (sumanthratna) [inline doc/source/tune/why_tune.rst:15]: nit: "logging results to tools" (note the lack of "of")

File: doc/source/tune.rst [hunk 4]
```diff
@@ -52,9 +51,16 @@ If using TF2 and TensorBoard, Tune will also automatically generate TensorBoard
 
 .. tip:: Join the `Ray community slack <https://forms.gle/9TSdDYUgxYs8SA9e8>`_ to discuss Ray Tune (and other Ray libraries)!
 
-Guides/Materials
+Why choose Tune?
 ----------------
 
+There are many other hyperparameter optimization libraries out there. If you're new to Tune, you're probably wondering what makes Tune different?
```
Comment (sumanthratna) [inline doc/source/tune.rst:57]: nit: If you're new to Tune, you're probably wondering, "what makes Tune different?"

**Review by sumanthratna — COMMENTED** [review]
I like it! I left 3 very insignificant and annoying comments about wording, but IMO this does a good job of explaining why users should use Tune.

File: doc/source/tune/_tutorials/overview.rst [hunk 5]
```diff
@@ -176,3 +176,15 @@ Contributed Examples
...
+
+ - `Softlearning <https://github.com/rail-berkeley/softlearning>`_: Softlearning is a reinforcement learning framework for training maximum entropy policies in continuous domains. Includes the official implementation of the Soft Actor-Critic algorithm.
+ - `Flambe <https://github.com/asappresearch/flambe>`_: An ML framework to accelerate research and its path to production. See `flambe.ai <flambe.ai>`_.
+ - `Population Based Augmentation <https://github.com/arcelien/pba>`_: Population Based Augmentation (PBA) is a algorithm that quickly and efficiently learns data augmentation functions for neural network training. PBA matches state-of-the-art results on CIFAR with one thousand times less compute.
+ - `Fast AutoAugment by Kakao <https://github.com/kakaobrain/fast-autoaugment>`_: Fast AutoAugment (Accepted at NeurIPS 2019) learns augmentation policies using a more efficient search strategy based on density matching.
+ - `Allentune <https://github.com/allenai/allentune>`_: Hyperparameter Search for AllenNLP from AllenAI.
+ - `machinable <https://github.com/frthjf/machinable>`_: A modular configuration system for machine learning research. See `machinable.org <machinable.org>`_.
```
Comment (richardliaw) [inline doc/source/tune/_tutorials/overview.rst:190]:
```suggestion
 - `machinable <https://github.com/frthjf/machinable>`_: A modular configuration system for machine learning research. See `machinable.org <https://machinable.org>`_.
```

File: doc/source/tune/_tutorials/overview.rst [hunk 6]
```diff
@@ -176,3 +176,15 @@ Contributed Examples
...
 - `genetic_example <https://github.com/ray-project/ray/blob/master/python/ray/tune/examples/genetic_example.py>`__: Optimizing the michalewicz function using the contributed GeneticSearch algorithm with AsyncHyperBandScheduler.
 - `tune_cifar10_gluon <https://github.com/ray-project/ray/blob/master/python/ray/tune/examples/tune_cifar10_gluon.py>`__: MXNet Gluon example to use Tune with the function-based API on CIFAR-10 dataset.
+
+Open Source Projects using Tune
+-------------------------------
+
+Here are some of the popular open source repositories and research projects that leverage Tune. Feel free to submit a pull-request adding (or requesting a removal!) of a listed project.
+
+ - `Softlearning <https://github.com/rail-berkeley/softlearning>`_: Softlearning is a reinforcement learning framework for training maximum entropy policies in continuous domains. Includes the official implementation of the Soft Actor-Critic algorithm.
+ - `Flambe <https://github.com/asappresearch/flambe>`_: An ML framework to accelerate research and its path to production. See `flambe.ai <flambe.ai>`_.
```
Comment (richardliaw) [inline doc/source/tune/_tutorials/overview.rst:186]:
```suggestion
 - `Flambe <https://github.com/asappresearch/flambe>`_: An ML framework to accelerate research and its path to production. See `flambe.ai <https://flambe.ai>`_.
```

File: doc/source/tune/why_tune.rst (hunk 2 above)
Comment (krfricke) [reply doc/source/tune/why_tune.rst:6]: The emphasis on *schedules* is a bit odd if we don't talk about it afterwards. I would just remove it. 
If we added bullet points, I would expect this in all sub chapters, so I it's good the way it is now.

File: doc/source/tune/why_tune.rst [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+Hyperparameter tuning is known to be highly time-consuming, so it is often necessary to parallelize this process. Most other tuning frameworks require you to implement your own multi-process framework or build your own distributed system to speed up hyperparameter tuning.
+
+However, Tune allows you to transparently :ref:`parallelize across multiple GPUs and multiple nodes <tune-parallelism>`. Tune even has seamless :ref:`fault tolerance and cloud support <tune-distributed>`, allowing you to scale up your hyperparameter search by 100x while reducing costs by up to 10x by using cheap preemptible instances.
+
+What if I'm already doing hyperparameter tuning?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You might be already using an existing hyperparameter tuning tool such as HyperOpt or Bayesian Optimization.
+
+In this situation, Tune actually allows you to power up your existing workflow. Tune's :ref:`Search Algorithms <tune-search-alg>` integrate with a variety of popular hyperparameter tuning libraries (such as Nevergrad or HyperOpt) and allow you to seamlessly scale up your optimization process -- without sacrificing performance.
```
Comment (krfricke) [inline doc/source/tune/why_tune.rst:30]: Maybe mention that existing hyperparameter configurations don't have to be changed at all.

**Review by krfricke — APPROVED** [review]

File: doc/source/tune/why_tune.rst (hunk 2 above)
Comment (richardliaw) [reply doc/source/tune/why_tune.rst:6]:
```suggestion
Tune enables you to leverage a variety of these cutting edge optimization algorithms, reducing the cost of tuning by `aggressively terminating bad hyperparameter evaluations <tune-scheduler-hyperband>`_, intelligently :ref:`choosing better parameters to evaluate <tune-search-alg>`, or even :ref:`changing the hyperparameters during training <tune-scheduler-pbt>` to optimize hyperparameter schedules.
```

**Pull request merged**